hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/Message.java

```diff
@@ -82,8 +86,9 @@ public boolean isTtlExceeded(long ttlMillis) {
         return currentTimestamp > readingTimestamp + ttlMillis;
     }
 
-    public void incrementRetryCounter() {
+    public void incrementRetryCounter(Collection<String> succeededUris) {
         this.retryCounter++;
+        this.succeededUris = ImmutableSet.copyOf(succeededUris);
```

Comment:
shouldn't this be an append operation? and i don't know if we want to create new collection for each retry, think about our poor allocation rate :) maybe appending to the set would be enough, sacrificing the internal immutability?